ballerina/natives.bal

```diff
@@ -123,6 +135,31 @@ public isolated function printWarn(string msg, error? 'error = (), *KeyValues ke
     }
 }
 
+# Set the log output to a file. Note that all the subsequent logs of the entire application will be written to this file.
+# ```ballerina
+# var result = log:setOutputFile("./resources/myfile.log");
+# var result = log:setOutputFile("./resources/myfile.log", log:OVERWRITE);
+# ```
+#
+# + path - The path of the file
+# + option - To indicate whether to overwrite or append the log output
+#
+# + return - A `log:Error` if an invalid file path was provided
+public isolated function setOutputFile(string path, FileWriteOption option = APPEND) returns Error? {
+    lock {
+        if !path.endsWith(".log") {
+            return error Error("The given path is not valid. Should be a file with .log extension.");
+        }
+        outputFilePath = path;
```

Comment:
Assigning to the `outputFilePath` should come last after all checks.

Reply:
Updated in 1a72ddc